go/subtle/util.go

```diff
@@ -85,11 +88,18 @@ func GetCurve(curve string) elliptic.Curve {
 }
 
 // ComputeHash calculates a hash of the given data using the given hash function.
-func ComputeHash(hashFunc func() hash.Hash, data []byte) []byte {
+func ComputeHash(hashFunc func() hash.Hash, data []byte) (hashed []byte, err error) {
+	if hashFunc == nil {
```

Comment:
A nil function would fail anyway, I don't think this check is necessary.

Reply:
We have to check nil, other wise calling `hashFunc()` below if `hashFunc` is nil will panic